W-10957252-MultipleIDPs-ESB (#256)
mulesoft/docs-composer: en/mulesoft/modules/ROOT/pages/ms_composer_prerequisites.adoc

A message appears, warning you that deleting a user cannot be undone.
. Click *Delete*.

=== Enable Multiple Identification Providers (IDP)

As an administrator, you can enable users to add identity providers (IDPs) for their organization. Note that:

> * If an IDP user is deleted from MuleSoft Composer and that user subsequently logs in to MuleSoft Composer using the IDP, the user profile is restored in the `Disabled` state. To allow the user to authenticate again, an Admin can toggle the state to `Enabled`.

Reviewer: ```suggestion
* If an IDP user is deleted from MuleSoft Composer and that user subsequently logs in to MuleSoft Composer using the IDP, the user profile is restored in the `Disabled` state. To allow the user to authenticate again, an toggle the state to `Enabled`.
```

> * Admins cannot disable the creation of new users via the *Users* page.

Reviewer: ```suggestion
* You cannot disable the creation of new users via the *Users* page.
```

* Deleting an IDP user does not stop the flows activated by that user.

> * When MuleSoft Composer is dynamically registered as an OpenID Connect SSO provider in Okta, the MuleSoft Composer app is named “Anypoint Platform” in Okta.

Reviewer: ```suggestion
* When MuleSoft Composer is dynamically registered as an OpenID Connect SSO provider in Okta,  MuleSoft Composer is named “Anypoint Platform” in Okta.
```

For more information on identity management in MuleSoft, see the xref:access-management::external-identity.adoc[Access Management] documentation.

To enable OpenID Connect (OIDC) IDP:

. In the MuleSoft Composer sidebar, click *Settings* and then click *Multiple IDP*.
. On the *Identity Provider* page, click *Add Identity Provider* > *OpenID Connect*.
. In the *New Identity Provider* page, complete the required fields:
+
* *Name*: The nickname for this IDP.
* *Client Registration URL*: This field appears when *Dynamic Registration* is selected. The URL to dynamically register client applications for your identity provider.

> * *Authentication Header*: This field appears when *Dynamic Registration* is selected. The header provides credentials to authenticate a server. This header is required if the provider restricts registration requests to authorized clients.

Reviewer: ```suggestion
* *Authentication Header*:  The header that provides credentials to authenticate a server. This header is required if the provider restricts registration requests to authorized clients. This field appears when you select *Dynamic Registration*.
```

> * *Client ID*: This field appears when *Manual Registration* is selected. The unique identifier that you provided for your manually created client application.

Reviewer: Can you make this change about the field appearing GLOBALLY

* *Client Secret*: This field appears when *Manual Registration* is selected. The password, or secret, for authenticating your Composer organization with your Identity Provider.

> * *OpenID Connect Issuer*: The location of the OpenID Provider. For most providers, `/.well-known/openid-configuration` is appended to the issuer to generate the metadata URL for openID Connect specifications. For Salesforce, you must provide the value for `issuer`.

Reviewer: Confirm that it should say "For Salesforce..."

* *Authorize URL*: The URL where the user authenticates and grants OpenID Connect client applications access to the user's identity.
* *Token URL*: The URL that provides the user’s identity encoded in a secure JSON Web Token.

> * *User Info URL*: URL that returns user profile information to the client app.

Reviewer: ```suggestion
* *User Info URL*: The URL that returns user profile information to the client app.
```

. Optionally, expand *Advanced Settings*, and provide the following values:
+
* *Group Scope*: The OIDC scope to request the group claim.
* *Group Attribute JSON Data Expression*: The JSONata expression used to select the groups from the user information or ID token. The result must be an array of strings.
* *Disable server certificate validation* checkbox: Select checkbox to disable server certificate validation if your OpenID client management instance presents a self-signed certificate or one signed by an internal certificate authority.

. Click *Save*.
. Log out of Composer, navigate to the sign-on URL you entered in the *New Identity Provider* page, and then log in through your identity provider to test the configuration.

To enable SAML 2.0 IDP:

Note that the file-based configuration of a SAML 2.0 is not supported.

. In the MuleSoft Composer sidebar, click *Settings* and then click *Multiple IDP*.
. On the *Identity Provider* page, click *Add Identity Provider* > *SAML 2.0*.
. In the *New Identity Provider* page, complete the required fields:
+
* *Name*: Enter a nickname for this IDP.

> * *Sign On URL*: Redirect URL provided by the IDP for sign in. For example, `https://example.com/sso/saml`.

Reviewer: ```suggestion
* *Sign On URL*: The redirect URL provided by the IDP for sign in. For example, `https://example.com/sso/saml`.
```

> * *Sign Off URL*: URL to redirect signout requests, so users both sign out of Composer and have their SAML user’s status set to `signed out`.

Reviewer: ```suggestion
* *Sign Off URL*: The URL to redirect signout requests, so users both sign out of Composer and have their SAML user’s status set to `signed out`.
```

Reviewer: Sign out or signout? Think it should be sign out

> * *Issuer*: ID of the identity provider instance that sends SAML assertions.

Reviewer: ```suggestion
* *Issuer*: The ID of the identity provider instance that sends SAML assertions.
```

> * *Public Key*: Public key provided by the identity provider, which is used to sign the SAML assertion. It is the `X509Certificate` value in the SAML Response.

Reviewer: ```suggestion
* *Public Key*: The Public key provided by the identity provider, which is used to sign the SAML assertion. It is the `X509Certificate` value in the SAML Response.
```

* *Audience*: An arbitrary string value that identifies your Composer organization. The typical value for this string is `<organizationDomain>.composer.mulesoft.com`.
* *Single Sign On Initiation*: Specify whether SSO can be initiated by Composer, your identity provider (for example, Okta), or both.

** The *Service Provider Only* option allows only the Composer to initiate SSO.
** The *Identity Provider Only* option allows only your external identity provider to initiate SSO.
** The *Both* option allows either Composer or your external identity provider to initiate SSO. +

> The default value for this setting for newly configured identity provider configurations is Both.

Reviewer: If this is UI should Both be formatted in bold?

> The default value for this setting for newly configured identity provider configurations is Both.

Reviewer: ```suggestion
The default value for this setting for newly configured identity provider configurations is **Both**.
```

. Optionally, expand *Advanced Settings*, and provide the following values:
+

> * *Username Attribute*: Field name in the SAML `AttributeStatements` that maps to username. If no value is configured, the `NameID` attribute of the SAML `Subject` is used (Note: this is outside the SAML `AttributeStatements`).

Reviewer: ```suggestion
* *Username Attribute*: Field name in the SAML `AttributeStatements` that maps to the user's name. If no value is configured, the `NameID` attribute of the SAML `Subject` is used (Note: this is outside the SAML `AttributeStatements`).
```

or `username`?

* *First Name Attribute*: Field name in the SAML `AttributeStatements` that maps to `First Name`.
* *Last Name Attribute*: Field name in the SAML `AttributeStatements` that maps to `Last Name`.
* *Email Attribute*: Field name in the SAML `AttributeStatements` that maps to `Email`.
* *Group Attribute*: Field name in the SAML `AttributeStatements` that maps to `Group`.

> * *Require encrypted SAMl assertions* checkbox: If enabled, the SAML assertions sent from the IdP must be encrypted and follow the guidelines mentioned in the Prerequisites.

Reviewer: ```suggestion
* *Require encrypted SAMl assertions* checkbox: If enabled, the SAML assertions sent from the IDP must be encrypted and follow the guidelines mentioned in the prerequisites.
```

. Click *Save*.

> . Log out of Composer, navigate to the sign-on URL you entered in the *New Identity Provider* page, and then log in through your identity provider to test the configuration.

Reviewer: Confirm that you can just say Composer, or does it need to be MuleSoft Composer?
